src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsService.kt

```diff
@@ -189,6 +189,8 @@ class AppointmentsService(
         "${instance.appointmentDescription} - ${instance.comment}"
       }
     }
+    return comment?.run { if (length <= 4000) this else substring(0, 4000) }
```

Comment:
very minor but `let` is probably the correct semantics, but either way I think you can do 

`return comment?.take(4000)`

(I thing that is the function name)